aten/src/ATen/SparseTensorImpl.h

```diff
@@ -69,9 +69,10 @@ struct SparseTensorImpl : public TensorImpl {
   // TODO: Figure out a more safe way to provide this functionality
   std::vector<int64_t>& _sizes_mut() { return size_; }
 
-  // WARNING: This function does NOT preserve invariants of sparseDims/denseDims with
+  // NOTE: This function checks and preserves invariants of sparseDims/denseDims with
   // respect to indices and values
   void raw_resize_(int64_t sparseDims, int64_t denseDims, ArrayRef<int64_t> size) {
+    #ifndef USE_TH_SIZE_ZERO_DIM
```

Comment:
is this function always called for resize operations?  Which ones call directly into here vs other functions (are there other functions)?

Reply:
I looked around and `raw_resize_` is the only function that can change the size of the sparse tensor (`_sizes_mut` exposes the underlying `size_` vector in a mutable way, but its usage is very limited).